docs/sphinx/user_guide/feature/resource.rst

```diff
@@ -0,0 +1,266 @@
+ Resource Policy Type
+ ======== ==============================
+ Cuda     | cuda_exec
+          | cuda_exec_async
+ Hip      | hip_exec
+          | hip_exec_async
+ Omp*     | omp_target_parallel_for_exec
+          | omp_target_parallel_for_exec_n
+ Host     | loop_exec
+          | seq_exec
+          | openmp_parallel_exec
+          | omp_for_schedule_exec
+          | omp_for_nowait_schedule_exec
+          | simd_exec
+          | tbb_for_dynamic
+          | tbb_for_static
+ ======== ==============================
+
+.. note:: * The ``RAJA::resources::Omp`` resource is still under development.
+
+IndexSet policies require two execution policies to define them. Currently, a users only need to pass a
+single resource to the forall Indexset call. This resource will be used to execute the inner 
+execution loop of the indexset policy.::
+
+    using ExecPol = RAJA::ExecPolicy<RAJA::seq_segit, RAJA::cuda_exec<256>>;
+    RAJA::forall<ExecPol>(my_cuda_res, iset,  .... );
+
+When a resource is not provided by the user, a *default* resource is assigned from within the RAJA
+forall implementation. This default resource can be accessed in a number of ways.
+
+Directly from the concrete resource type::
+
+    RAJA::resources::Cuda my_default_cuda = RAJA::resources::Cuda::get_defualt();
+
+The resource type can be deduced from an execution policy::
+
+    using Res = RAJA::resources::get_resource<ExecPol>::type;
+    Res r = Res::get_defualt();
+
+Deduced from an execution policy and return the default directly::
+
+    auto my_resource = RAJA::resources::get_default_resource<ExecPol>();
+
+.. note:: * For Cuda and Hip the default resource is *NOT* the CUDA or HIP default stream it is it's 
+            own stream defined in ``camp/include/resource/``. This is in an attempt to break away
+            from some of the issues that arise from the synchronization behaviour of the CUDA/HIP 
+            default stream. It is still possible to use the CUDA/HIP defined default stream as the
+            default resource. This can be enabled by defining ``CAMP_USE_PLATFORM_DEFAULT_STREAM``
+
+------
+Events
+------
+
+Event objects are a feature that allow users to wait or query the status of a resource's action. An
+event can be returned from a resource with::
+
+    RAJA::resources::Event e = my_res.get_event();
+
+Getting an event like this enqueues an event type object for the given backend. 
+
+You can call a blocking function and wait for that event::
+
+    e.wait();
+
+Preferably, users can enqueue the event to a specific resource, forcing only that resource to wait for the event::
+
+    my_res.wait_for(&e);
+
+The latter is useful as it allows the user to set up dependencies between resource objects and 
+``RAJA::forall`` calls.
+
+.. note:: *An Event object is only generated if a user specifically returns one from a ``RAJA::forall``::
+           call. This stops unnecessary event objects being created and causing performance hits when not
```

Comment:
This sentence makes sense, but "This stops ... causing performance hits" sounds awkward.

Reply:
Removed the "causing performace hits".